core/common/boot/src/main/java/org/eclipse/edc/boot/system/ExtensionLoader.java

```diff
@@ -127,4 +128,22 @@ public <T> List<T> loadExtensions(Class<T> type, boolean required) {
         return serviceLocator.loadImplementors(type, required);
     }
 
+    /**
+     * Parses the ConsoleMonitor log level from the program args. If no log level is provided, defaults to Level default.
+     */
+    private static ConsoleMonitor.Level parseLogLevel(String[] programArgs) {
+        return Set.of(programArgs).stream()
+                .filter(arg -> arg.startsWith(ConsoleMonitor.LEVEL_PROG_ARG + "="))
+                .map(arg -> arg.split("=").length == 2 ? arg.split("=")[1] : "null" )
```

Comment:
nit: adding another `map(arg -> arg.split("=")` before this avoids having to do it twice. 

Also, providing `"null"` seems spurious. Better to provide `null` as default

Reply:
[3b1253a](https://github.com/eclipse-edc/Connector/pull/4476/commits/3b1253a2a5d21ca8c584f8abf296ed69005e8727)